pennylane/fermi/conversion.py

```diff
@@ -49,6 +55,7 @@ def jordan_wigner(
         wire_map (dict): a dictionary defining how to map the oribitals of
             the Fermi operator to qubit wires. If None, the integers used to
             order the orbitals will be used as wire labels. Defaults to None.
+        tol (float): tolerance for discarding the imaginary part of the coefficients
```

Comment:
@soranjh is there a usecase where we expect an imaginary part? I'm just wondering whether we need to expose the tolerance to users?

Reply:
Yes, the example provided in the following has imaginary coefficients. We can make the tolerance user-facing for functions that build a Hamiltonian, but I think it just makes the signature for those functions more complicated.

Reply:
What is the default for `tol = None`?

Reply:
I intentionally did not provide a default value. If we have a default, then we automatically cast complex objects to real ones if the imaginary components is smaller than the default. I thought some users might get confused or might not want that. Currently, if there is no `tol` defined, nothing happens even if you have coefficients with zero imaginary values.

Reply:
Yes maybe we change the default to zero, which would also resolve some of the comments below.